File: frontend-react/src/api/me.ts
import apiClient from './client';

export interface MeResponse {
  principal_type: string;
  principal_id: string;
  display_name: string;
  email?: string | null;
  roles: string[];
  scopes: string[];
}

export async function getMe(): Promise<MeResponse> {
  const response = await apiClient.get<MeResponse>('/api/v1/me');
  return response.data;
}

File: backend/api/routes/me.py
"""Authenticated identity endpoint."""

from __future__ import annotations

from fastapi import APIRouter, Request

from backend.api.principal import principal_from_request_state

router = APIRouter()


@router.get("/me", tags=["identity"])
async def me(request: Request) -> dict[str, object]:
    principal = principal_from_request_state(request)
    return {
        "principal_type": principal.type,
        "principal_id": principal.id,
        "display_name": principal.display_name,
        "email": principal.email,
        "roles": sorted(principal.roles),
        "scopes": sorted(principal.scopes),
    }
